qa/rpc-tests/test_framework/util.py

```diff
@@ -125,12 +125,16 @@ def sync_blocks(rpc_connections, wait=1, timeout=60):
     """
     Wait until everybody has the same tip
     """
+    maxheight = 0
     while timeout > 0:
-        tips = [ x.getbestblockhash() for x in rpc_connections ]
+        tips = [ x.waitforblockheight(maxheight, int(wait * 1000)) for x in rpc_connections ]
+        heights = [ x["height"] for x in tips ]
         if tips == [ tips[0] ]*len(tips):
             return True
-        time.sleep(wait)
+        if heights == [ heights[0] ]*len(heights): #heights are the same but hashes are not
+            raise AssertionError("Block sync failed")
```

Comment:
There's a slightly bigger issue that @morcos just noticed, which is that the block notification signals doesn't quite work with invalidateblock right now.

I think we can work around it by having the invalidateblock rpc handler explicitly call the ui signals in this case (which is probably the right thing to do in general).